components/user_settings/notifications/desktop_notification_setting/desktop_notification_settings.tsx

```diff
@@ -26,34 +51,35 @@ export default class DesktopNotificationSettings extends React.PureComponent {
         this.dropdownSoundRef = React.createRef();
     }
 
-    handleMinUpdateSection = (section) => {
+    handleMinUpdateSection = (section: string): void => {
         this.props.updateSection(section);
-
         this.props.cancel();
     }
 
-    handleMaxUpdateSection = (section) => {
-        this.props.updateSection(section);
-    }
+    handleMaxUpdateSection = (section: string): void => this.props.updateSection(section);
 
-    handleOnChange = (e) => {
+    handleOnChange = (e: ChangeEvent<HTMLInputElement>): void => {
         const key = e.currentTarget.getAttribute('data-key');
         const value = e.currentTarget.getAttribute('data-value');
-        this.props.setParentState(key, value);
+        if (key && value) {
+            this.props.setParentState(key, value);
+        }
     }
 
-    handleThreadsOnChange = (e) => {
+    handleThreadsOnChange = (e: ChangeEvent<HTMLInputElement>): void => {
         const value = e.target.checked ? NotificationLevels.ALL : NotificationLevels.MENTION;
         this.props.setParentState('desktopThreads', value);
     }
 
-    setDesktopNotificationSound = (selectedOption) => {
-        this.props.setParentState('desktopNotificationSound', selectedOption.value);
-        this.setState({selectedOption});
-        Utils.tryNotificationSound(selectedOption.value);
+    setDesktopNotificationSound: ReactSelect['onChange'] = (selectedOption: ValueType<SelectedOption>): void => {
+        if (selectedOption && 'value' in selectedOption) {
```

Comment:
Did we had to add this if condition?

Reply:
Yes, this was necessary to avoid Typescript complaining about `value` doesn't exist in selectedOption: I see similar usages in the codebase: One example

https://github.com/mattermost/mattermost-webapp/blob/e949d96a76995b7149d7b467074f99bd40968db7/components/activity_and_insights/insights/time_frame_dropdown/time_frame_dropdown.tsx#L49-L53